### DISCUSSION TITLE
Importing existing citation keys/recovering old citation key formula

### DISCUSSION THREAD

**Discussion by alhom** [retorquere/zotero-better-bibtex#2729] [Q&A]
Hello,

After updating my laptop, I noticed I did not migrate Better BibTeX config. Now I have the issue of having ~2MB of an exported library with citation keys produced by apparent legacy formula of the form of `<author>_<first word of title>_year`, and fresh Better BibTex has a new formula for these.

My questions are then:
1) Can I import old citation keys en-masse from the old Better BibTeX export, onto the Zotero database that was re-acquired from the Zotero cloud, without citation keys?
2) How could I find the legacy citation key formulas for previous version to reproduce/approximate the old scheme?

**Comment by retorquere** [comment]
You do still have access to your old database? What OS are you on?

**Reply by alhom** [reply]
Yes I do have the old database. Running on Ubuntu.

**Reply by retorquere** [reply]
I'm working on a method to import the settings and keys.

**Reply by retorquere** [reply]
Please open an issue, I can drop trial builds there.

**Reply by alhom** [reply]
Thank you! Issue opened at https://github.com/retorquere/zotero-better-bibtex/issues/2733, but really not sure how a debug log ID would apply here ^^'.

I do also have some json config files from the old install, but I did not find my then-citation-key-formula from there, at least without any pointers for what to look for (which is also why I'm thinking I was using an old default).

**Reply by retorquere** [reply]
> Thank you! Issue opened at https://github.com/retorquere/zotero-better-bibtex/issues/2733, but really not sure how a debug log ID would apply here ^^'.


It would give me your current settings and citation keys before we start meddling with them so that I could see the effects of what we're doing. I *really* wish people would stop thinking they know better than me on this subject. I can sort of see what you think it's funny but I'm pretty tired of arguing with people about this. 

> I do also have some json config files from the old install, but I did not find my then-citation-key-formula from there, at least without any pointers for what to look for (which is also why I'm thinking I was using an old default).

I don't know what json config files are but they don't apply here.
